blocks/components/editable/dom-to-react.js

```diff
@@ -0,0 +1,41 @@
+import { createElement } from 'react';
+import { map, reduce } from 'lodash';
+import camelCaseAttrMap from 'html-to-react/lib/camel-case-attribute-names';
+
+export function nodeToReact( node, index ) {
+	if ( ! node ) {
+		return null;
+	}
+
+	if ( node.nodeName === '#text' ) {
+		return node.nodeValue;
+	}
+
+	if ( node.getAttribute( 'data-mce-bogus' ) === 'all' ) {
```

Comment:
So right now you can pass the createElement callback, which means that you can filter stuff, there is no dependency on React, and you can use it with other React-like libraries.